spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/BaseCatalog.java

```diff
@@ -46,4 +52,30 @@ public Procedure loadProcedure(Identifier ident) throws NoSuchProcedureException
 
     throw new NoSuchProcedureException(ident);
   }
+
+  @Override
+  public Identifier[] listFunctions(String[] namespace) throws NoSuchNamespaceException {
+    if (namespace.length == 1 && namespace[0].equalsIgnoreCase("system")) {
+      return SparkFunctions.list().stream()
+          .map(name -> Identifier.of(namespace, name))
+          .toArray(Identifier[]::new);
+    }
+
+    throw new NoSuchNamespaceException(namespace);
```

Comment:
This should return an empty list if the database exists.

Reply:
Fixed